gctscript/modules/gct/exchange.go

```diff
@@ -0,0 +1,329 @@
+	}
+
+	data := make(map[string]objects.Object, 2)
+	data["exchange"] = &objects.String{Value: rtnValue.Exchange}
+	data["currencies"] = &funds
+
+	return &objects.Map{
+		Value: data,
+	}, nil
+}
+
+// ExchangeOrderQuery query order on exchange
+func ExchangeOrderQuery(args ...objects.Object) (ret objects.Object, err error) {
+	if len(args) != 2 {
+		err = objects.ErrWrongNumArguments
+		return
+	}
+
+	exchangeName, _ := objects.ToString(args[0])
+	orderID, _ := objects.ToString(args[1])
+
+	orderDetails, err := modules.Wrapper.QueryOrder(exchangeName, orderID)
+	if err != nil {
+		return
+	}
+
+	var tradeHistory objects.Array
+	for x := range orderDetails.Trades {
+		temp := make(map[string]objects.Object, 7)
+		temp["timestamp"] = &objects.Time{Value: orderDetails.Trades[x].Timestamp}
+		temp["price"] = &objects.Float{Value: orderDetails.Trades[x].Price}
+		temp["fee"] = &objects.Float{Value: orderDetails.Trades[x].Fee}
+		temp["amount"] = &objects.Float{Value: orderDetails.Trades[x].Amount}
+		temp["type"] = &objects.String{Value: orderDetails.Trades[x].Type.String()}
+		temp["side"] = &objects.String{Value: orderDetails.Trades[x].Side.String()}
+		temp["description"] = &objects.String{Value: orderDetails.Trades[x].Description}
+		tradeHistory.Value = append(tradeHistory.Value, &objects.Map{Value: temp})
+	}
+
+	data := make(map[string]objects.Object, 14)
+	data["exchange"] = &objects.String{Value: orderDetails.Exchange}
+	data["id"] = &objects.String{Value: orderDetails.ID}
+	data["accountid"] = &objects.String{Value: orderDetails.AccountID}
+	data["currencypair"] = &objects.String{Value: orderDetails.CurrencyPair.String()}
+	data["price"] = &objects.Float{Value: orderDetails.Price}
+	data["amount"] = &objects.Float{Value: orderDetails.Amount}
+	data["amountexecuted"] = &objects.Float{Value: orderDetails.ExecutedAmount}
+	data["amountremaining"] = &objects.Float{Value: orderDetails.RemainingAmount}
+	data["fee"] = &objects.Float{Value: orderDetails.Fee}
+	data["side"] = &objects.String{Value: orderDetails.OrderSide.String()}
+	data["type"] = &objects.String{Value: orderDetails.OrderType.String()}
+	data["date"] = &objects.String{Value: orderDetails.OrderDate.String()}
+	data["status"] = &objects.String{Value: orderDetails.Status.String()}
+	data["trades"] = &tradeHistory
+
+	return &objects.Map{
+		Value: data,
+	}, nil
+}
+
+// ExchangeOrderCancel cancels order on requested exchange
+func ExchangeOrderCancel(args ...objects.Object) (ret objects.Object, err error) {
+	if len(args) != 2 {
+		err = objects.ErrWrongNumArguments
+		return
+	}
+
+	exchangeName, _ := objects.ToString(args[0])
+	orderID, _ := objects.ToString(args[1])
+
+	rtn, err := modules.Wrapper.CancelOrder(exchangeName, orderID)
+	if err != nil {
+		return
+	}
+
+	if rtn {
+		return objects.TrueValue, nil
+	}
+	return objects.FalseValue, nil
+}
+
+// ExchangeOrderSubmit submit order on exchange
+func ExchangeOrderSubmit(args ...objects.Object) (ret objects.Object, err error) {
```

Comment:
I don't think you should use declared return variables in these functions. The function is too long/complex and you also don't use `ret`

Reply:
Yeah I can change them my first response is normally to use name returns and work from there but going back over i agree not needed in some of these